libraries/AP_RangeFinder/AP_RangeFinder_RDS02UF.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * RDS02UF Note:
+ * Sensor range scope 1.5m~20.0m
+ * Azimuth Coverage ±17°,Elevation Coverage ±3°
+ * Frame Rate 20Hz
+ */
+
+#include "AP_RangeFinder_config.h"
+
+#if AP_RANGEFINDER_RDS02UF_ENABLED
+
+#include "AP_RangeFinder_RDS02UF.h"
+#include <AP_HAL/AP_HAL.h>
+
+#define RDS02_HEAD                  0x55
+#define RDS02_END                   0xAA
+#define RDS02UF_PRE_DATA_LEN        6
+#define RDS02_TARGET_INFO_FC        0x070C
+#define RDS02UF_IGNORE_ID_BYTE      0x0F0F
+#define RDS02UF_UAV_PRODUCTS_ID     0x03FF
+#define RDS02UF_IGNORE_CRC          0xFF
+#define RDS02UF_NO_ERR              0x00
+
+bool AP_RangeFinder_RDS02UF::get_reading(float &distance_m)
+{
+    if (uart == nullptr) {
+        return false;
+    }
+
+    const uint32_t nbytes = uart->read(&u.parse_buffer[body_length],
+                                 ARRAY_SIZE(u.parse_buffer)-body_length);
+    if (nbytes == 0) {
+        return false;
+    }
+    body_length += nbytes;
+
+    move_header_in_buffer(0);
+
+    // header byte 1 is correct.
+    if (body_length < ARRAY_SIZE(u.parse_buffer)) {
```

Comment:
```suggestion
    if (body_length < sizeof(u.packet)) {
```

... this is, after all, exactly what you're interested in.

The parse buffer can be larger than the packet, allowing for multiple packets to be read in at once (and different packets to be parsed out of the same buffer, of course)